### ISSUE TITLE
Move misplaced mk_restarts directory; clean up duplication of ReadCNTilFile(), ReadTileFile_RealLatLon() etc,  

### ISSUE THREAD

**Issue by gmao-rreichle** [GEOS-ESM/GEOSgcm_GridComp#232]
Issue #225 and pull request #224 addressed some cleanup of routines in the "mk_restarts" directory, including a renaming of subroutine ReadCNTilFile() to ReadTileFile_RealLatLon() for clarity.
This issue assumes that pull request #224 has been merged.
There are two additional concerns:

1) A few subroutines, incl. to_radian(), haversine() and ReadCNTilFile() a.k.a. ReadTileFile_RealLatLon() 
appear to be duplicated in mk_restarts/getids.F90 and GEOSsurface_GridComp/Shared/Raster/comp_CATCHCN_AlbScale_parameters.F90

2) Subroutine "to_radian()" should probably be eliminated in favor of the corresponding MAPL routine.

3) It is not clear why "mk_restarts" lives in GEOScatch_GridCom/ when it addresses re-tiling for other (non-catch) GridComps.  My feeling is that some reorganization would be needed.

@sdrabenh suggest that @tclune comment on this on behalf of the SI team to see what should be done.

**Comment by gmao-rreichle** [comment]
More duplication:
Subroutine calc_soil_moist() is defined in mk_restarts/Scale_Catch.F90 and also in mk_restarts/Scale_CatchCN.F90, both of which duplicate the functionality of subroutine catch_calc_soilmoist() in GEOSland_GridComp/Shared/lsm_routines.F90
(Note also subroutines partition() and rzequil() in Scale_Catch[CN].F90.)

Re. item 3 above (code reorganization):
E.g., subroutine Scale_CatchCN.F90 is in the GEOS**catch**_GridComp/ subdirectory but should probably be in the Catch**CN** Grid Comp dir.

@gmao-jkolassa @biljanaorescanin @rdkoster @weiyuan-jiang

**Comment by gmao-jkolassa** [comment]
I am happy to do clean up Scale_Catch.F90 and Scale_CatchCN.F90, but I think it makes sense to first move those routines to their final locations, just so that dependencies can be established correctly. 

I had been discussing with Rolf via email about possibly creating a mk_restarts directory in GEOScatchCN_GridComp and I have included his response below:

> Maybe it makes more sense to keep "mk_restarts" intact as a single dir but place it at the level of the Surface GridComp?  Or, if we were to go with separate "mk_restarts" directories in the Catch, CatchCN, etc GridComps, any files or subroutines that are common to more than one GridComp would need to be moved to a higher-level shared directory.

Are there any preferences or advantages to the proposed locations for the mk_restarts directory? Once we have agreed on a location, would you mind actually moving the files/directories @weiyuan-jiang ?

**Comment by biljanaorescanin** [comment]
Just a short summary of email communication we had so far on move of  the **mk_restarts** directory to the **GMAO_Shared**. 

The main issue we see is this from the mk_restarts CMakeLists.txt:
foreach (src ${exe_srcs})
  string (REGEX REPLACE ".F90" "" exe ${src})
  ecbuild_add_executable (
    TARGET ${exe}
    SOURCES ${src}
    LIBS MAPL gftl-shared GEOS_SurfaceShared GEOSroute_GridComp GEOS_LandShared ${this})
endforeach ()

If mk_restarts is moved to GMAO_Shared where we wanted, agreed at Model Development meeting to move, then we can have a circular dependency. We know that parts of GEOSgcm_GridComp depends on GMAO_Shared, but now GMAO_Shared would depend on GEOSgcm_GridComp. Nothing in GMAO_Shared depends on GEOSgcm_GridComp at the moment, as far as we know.

We also plan to wait on current PR under GEOSgcm_GridComp to be resolved https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/368 to proceed with this move since that PR will add one more restart for land CatchCN CLM 4.5 to this directory.

From other communication with @sanAkel  they wish to move their ocean/sea ice stuff off of mk_restarts in any case. 
Here is the list:
1. SaltImpConverter.F90
2. SaltIntSplitter.F90
3. cv_SaltRestart.F90
4. mk_CICERestart.F90
5. mk_RouteRestarts.F90
And the following 1-4 are for ocean ("salt") and sea-ice  ("CICE") and have nothing to do with land.
mk_RouteRestarts.F90 is new and still under development to my understanding.

I think this is the list of all interested or involved sides I apologize if I missed someone: 
@weiyuan-jiang @gmao-rreichle @gmao-jkolassa @sanAkel @mathomp4 @smahanam

**Comment by gmao-rreichle** [comment]
Quick correction:  "mk_RouteRestarts.F90" is part of the land -- it addresses the runoff routing (streamflow) module that is under development.

**Comment by sanAkel** [comment]
@yvikhlya @zhaobin74 and myself will work to get 
```
1. SaltImpConverter.F90
2. SaltIntSplitter.F90
3. cv_SaltRestart.F90
4. mk_CICERestart.F90
```
out of `mk_restarts` FYI: @GEOS-ESM/cmake-team

**Comment by yvikhlya** [comment]
> @yvikhlya @zhaobin74 and myself will work to get
> 
> ```
> 1. SaltImpConverter.F90
> 2. SaltIntSplitter.F90
> 3. cv_SaltRestart.F90
> 4. mk_CICERestart.F90
> ```
> 
> out of `mk_restarts` FYI: @GEOS-ESM/cmake-team

But these are actually utils to create surface restarts. Should the whole mk_restarts be moved to GEOSsurface_GridComp/Shared instead? @tclune @mathomp4

**Comment by sanAkel** [comment]
@yvikhlya please ⏫ https://github.com/GEOS-ESM/GEOSgcm_GridComp/issues/232#issuecomment-768495703

**Comment by smahanam** [comment]
> But these are actually utils to create surface restarts. Should the whole mk_restarts be moved to GEOSsurface_GridComp/Shared instead? @tclune @mathomp4

Yuri has a valid point.

 I am not recommending anything over a something that has been discussed at length and decided already.
 This is merely my observation too, GEOSsurface_GridComp/Shared is not a bad place for this mk_restarts directory. 'Raster" can be renamed to "mk_surface_bcs" or something.

**Comment by zhaobin74** [comment]
I agree  with Yury. The ocean/sea ice regridding should stay with the others.

**Comment by yvikhlya** [comment]
> @yvikhlya please ⏫ [#232 (comment)](https://github.com/GEOS-ESM/GEOSgcm_GridComp/issues/232#issuecomment-768495703)

Biljana is talking about difficulties to move mk_restarts to GMAO_Shared, but GEOSsurf_GridComp has its own Shared directory and as far as I remember everything under mk_restarts is related to surface restarts.

**Comment by biljanaorescanin** [comment]
@smahanam nothing was decided. We proposed to re-organize so that individual restart routines are in more appropriate location( more intuitive to locate for user). Since we all agree on that currently mk_restarts is located in GEOSCatch_GridComp but contains many routines that are not related to Catchment. 
We proposed these two options at model development meeting:
1. Keep mk_restarts directory intact as it is, but move to a higher level folder, e.g. GEOSSurface_GridComp
2. Create mk_restarts directory in each GridComp subdirectory and move restarts routines to appropriate directory. 
I think we hoped people would like option 2. , like it would work for everyone but @sanAkel suggested moving it to the GEOS_Util. Which turned out to be in circular dependency then.

**Comment by yvikhlya** [comment]
No, GEOS_Util is not good location for mk_restarts.

**Comment by biljanaorescanin** [comment]
@yvikhlya @smahanam  It sounds like you both agree on moving it as it is but to the https://github.com/GEOS-ESM/GEOSgcm_GridComp/tree/develop/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared?  
@gmao-rreichle @weiyuan-jiang

**Comment by yvikhlya** [comment]
> @yvikhlya @smahanam It sounds like you both agree on moving it as it is but to the https://github.com/GEOS-ESM/GEOSgcm_GridComp/tree/develop/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared?
> @gmao-rreichle @weiyuan-jiang

Yes, I would suggest to do so if nobody objects.

**Comment by sanAkel** [comment]
> I agree with Yury. The ocean/sea ice regridding should stay with the others.

why should they
>should stay with the others.
?

**Comment by sanAkel** [comment]
> @smahanam nothing was decided. We proposed to re-organize so that individual restart routines are in more appropriate location( more intuitive to locate for user). Since we all agree that currently mk_restarts is located in GEOSCatch_GridComp but contains many routines that are not related to Catchment.
> We proposed these two options at model development meeting:
> 
> 1. Keep mk_restarts directory intact as it is, but move to a higher level folder, e.g. GEOSSurface_GridComp
> 2. Create mk_restarts directory in each GridComp subdirectory and move restarts routines to appropriate directory.
>    I think we hoped people would like option 2. , like it would work for everyone but @sanAkel suggested moving it to the GEOS_Util. Which turned out to be in circular dependency then.

@biljanaorescanin 

>but @sanAkel suggested moving it to the GEOS_Util. Which turned out to be in circular dependency then. 

This _suggestion_ is outdated! Please see https://github.com/GEOS-ESM/GEOSgcm_App/discussions/193 instead.

**Comment by zhaobin74** [comment]
> > I agree with Yury. The ocean/sea ice regridding should stay with the others.
> 
> @zhaobin74 why should they
> 
> > should stay with the others.
> > ?

because they are all children of surface

**Comment by sanAkel** [comment]
> > > I agree with Yury. The ocean/sea ice regridding should stay with the others.
> > 
> > 
> > @zhaobin74 why should they
> > > should stay with the others.
> > > ?
> 
> because they are all children of surface

where is the explicit connectivity? line numbers?

**Comment by yvikhlya** [comment]
> > > > I agree with Yury. The ocean/sea ice regridding should stay with the others.
> > > 
> > > 
> > > @zhaobin74 why should they
> > > > should stay with the others.
> > > > ?
> > 
> > 
> > because they are all children of surface
> 
> where is the explicit connectivity? line numbers?

These mk_*Restarts are not well separated. For example mk_LakeLaniceSaltRestarts makes lake, landice and saltwater restarts, so where would we put it?

**Comment by weiyuan-jiang** [comment]
@yvikhlya @smahanam , In a GridComp, usually the Shared directory is built first, then the children GridComp and the siblings which depends on the shared directory.  If you put all the mk_restart in Shared directories, there would be dependency problems. We either move all mk_restarts under SurfaceGridComp that depends on all the children and siblings or create a mk_restart directory under each grid comp.

**Comment by sanAkel** [comment]
>create a mk_restart directory under each grid comp.

sounds better.

**Comment by yvikhlya** [comment]
Then where mk_LakeLandiceSaltRestarts would go?

**Comment by sanAkel** [comment]
under "surface" following @weiyuan-jiang idea 
>under each grid comp

**Comment by zhaobin74** [comment]
Why does  landlice, lake and saltwater need  to have duplicated utilities if one works for them all?

**Comment by weiyuan-jiang** [comment]
I saw the dependencies in mk_restarts/CMakelists.txt here, but that may be false dependency. let me verify this first

foreach (src ${exe_srcs})
  string (REGEX REPLACE ".F90" "" exe ${src})
  ecbuild_add_executable (
    TARGET ${exe}
    SOURCES ${src}
    LIBS MAPL gftl-shared GEOS_SurfaceShared GEOSroute_GridComp GEOS_LandShared ${this})
endforeach ()

**Comment by yvikhlya** [comment]
I don't get how simply moving mk_restarts to GEOSsurf_GridComp/Shared would create dependency problems? Right now it is in a completely weird location and we don't have dependency problems. 

Creating separate mk_restarts under each surface child is a possible solution, but it looks to me a little more complex than just moving the whole thing.

**Comment by weiyuan-jiang** [comment]
We do have some dependency problems. For example, the file Scale_Catch.F90 depends on land_shared.  We cannot just move this program to GEOSsurf_GridComp/Shared  because GEOSland_GridComp/Shared depends on the upper level's Shared

**Comment by weiyuan-jiang** [comment]
But we can move everything to new GEOSsurf_GridComp/mk_restarts . That will be a very simple solution @yvikhlya

**Comment by smahanam** [comment]
I understand Weiyuan's difficulty, but we can ask Tom - he might have a suggestion. Maybe, 'Shared' is a wrong name for the content then, these codes are actually "Application" category - they don't share modules. They use others modules.
I just saw Weiyuan's message (but please put Raster and mk_restarts together inside a new directory)

- I vote to keep all land, landice, sea, seaice together in one directory.
- the land programs wish to stay inside GEOSgcm_GridComp repo for 2 reasons:
1) They have use statements to science modules and libraries because those codes are part of science.
2) Some science features sometimes  need upgrades to restart files, surface bcs files and scripts, in addition to the modifications that we make to to grid components, . It makes developer's job easy when everything in the same repo. For e.g. I have 3 or 4 active branches in GESOgcm_GridComp, I work in tandem in both in GEOS**_GridComp, mk_restarts in the same repo. 

- another question, shouldn't GEOS_Util/post/regrid.pl join the same directory should all stay together?

**Comment by yvikhlya** [comment]
> But we can move everything to new GEOSsurf_GridComp/mk_restarts . That will be a very simple solution @yvikhlya

I would support this.

**Comment by weiyuan-jiang** [comment]
@yvikhlya @smahanam,  We are closer and closer. How about we create GEOSsurf_GridComp/Utils which includes two sub directrories mk_restarts and Rasters?  I think we can move some files from Rasters to GEOSsurf_GridComp/Shared . The file GEOS_Util/post/regrid.pl can be moved to GEOSsurf_GridComp/Utils too.

**Comment by smahanam** [comment]
Weiyuan, sounds good to me - I don't involve in that kind of decisions, though.

> @zhaobin74 why should they
> 

Land wanst to keep an eye on other groups, for e.g., landice got caught red-handed once while encroaching on our prime real estate - land pixels. Somebody said AOGCM is encroaching our beaches. Those are the real questions that trouble us - just kidding.

**Comment by yvikhlya** [comment]
> @yvikhlya @smahanam, We are closer and closer. How about we create GEOSsurf_GridComp/Utils which includes two sub directrories mk_restarts and Rasters? I think we can move some files from Rasters to GEOSsurf_GridComp/Shared . The file GEOS_Util/post/regrid.pl can be moved to GEOSsurf_GridComp/Utils too.

regrid.pl regrids all restarts, not just surface actually. So, it probably belongs to some higher level component.

**Comment by sanAkel** [comment]
if you want to 👀 you can make yourself be included in "code owners" that's a request to @mathomp4

**Comment by smahanam** [comment]
> > @yvikhlya @smahanam, We are closer and closer. How about we create GEOSsurf_GridComp/Utils which includes two sub directrories mk_restarts and Rasters? I think we can move some files from Rasters to GEOSsurf_GridComp/Shared . The file GEOS_Util/post/regrid.pl can be moved to GEOSsurf_GridComp/Utils too.
> 
> regrid.pl regrids all restarts, not just surface actually. So, it probably belongs to some higher level component.

SURE ! If you say so,

**Comment by zhaobin74** [comment]
> > But we can move everything to new GEOSsurf_GridComp/mk_restarts . That will be a very simple solution @yvikhlya
> 
> I would support this.

second this

**Comment by weiyuan-jiang** [comment]
OK. Let me restructure and move around the mk_restarts first.  Not much work, so even if it is not approved, it is fine :-)

**Comment by sanAkel** [comment]
> OK. Let me restructure and move around the mk_restarts first. Not much work, so even if it is not approved, it is fine :-)

@tclune does this sound good to you?

**Comment by weiyuan-jiang** [comment]
Ok, folks, I have pushed a branch feature/wjiang/mk_restarts_re_org  to github that relocates mk_restarts and Raster.  Basically I add a sub-directory Utils under GEOSsurface_GridComp, and move Raster and mk_restarts under Utils.  As Yury suggested, we are better not to move regrid.pl at this point.  If you don't have objection,  I will talk to @tclune  and @mathomp4 about this change.

**Comment by gmao-rreichle** [comment]
Many thanks to everyone for the inspired discussion, and to @weiyuan-jiang for implementing a workable option.
Moving the whole package to Surface is far better than keeping it in Catch.  Putting the package into a new Surface/util directory should get us around the circular dependency that we would otherwise have with mk_Catch[CN]restarts in Surface/Shared or GEOS_Util.  (Sorry, Catch restarts are a pain...)
To move things to GEOS_Util is theoretically possible but it would require meticulously dissecting mk_Catch[CN]restarts, which would be considerably harder and for which we probably don't have the resources.  It would also be far riskier because there's no routine zero-diff test for mkRestarts.

**Comment by tclune** [comment]
I don't know what the right near-term location should be for the mk_restarts.   One thing I am considering in the longer term, is that as we establish a data repository (Git LSF) for model inputs, maybe these programs belong in that repository?      I'm interested in opinions.

The pro's of putting these in the data repository is that it makes it clear how the input data are created and what to change to generate new data.    OTOH,  these programs presumably evolve with the components that they support, which would argue to co-locate the program with the grid comp.

If the programs were truly standalone, I'd  definitely push for the former, but to the extent they need to link to GEOS libraries, the latter has appeal.

Am interested in all opinions.   Soon I'll be drafting a proposal for the data repository and give senior staff the opportunity to chime in.

**Comment by gmao-rreichle** [comment]
@tclune : Good to hear about the plans for a data repository.  That's attractive for many reasons, but it won't solve the fact that at least for Catch[CN] restarts, the bulk of the code that processes restarts needs to stay within the GCM GridComp repository.  For Catch[CN], creating restarts involves a good helping of science routines that live in the GCM GridComp, and it doesn't make sense to duplicate these in a different repository.  So we could at best move some top-level scripts into the data repository, but not the whole package.  What's more, the current mkRestarts package cannot be split easily into elements that remain in the GCM GridComp and elements that can be moved outside the repository.  Put differently, moving mkRestarts to a data repository is no easier than moving it to GEOS_Util.  We may find that Surface/util is the only realistic option for mkRestarts.
I don't know if the "Raster" s/w to create boundary conditions would split more easily.  @smahanam should be able to comment on this.  
For now, I think we should go ahead with Weiyuan's branch that moves mkRestarts and Raster to Surface/util.  That would definitely be an improvement over the status quo.  This doesn't mean we can't discuss further modifications, but those would take more time and effort.

**Comment by yvikhlya** [comment]
@tclune Managing versions of data files with LFS would be a great option imo. As for programs which generate data files, I am not sure where it is better to keep them. I guess, it will become more clear when LFS option is available. To generate a coupled model grid, we use a whole bunch of programs. Some of these programs are stand alone, but there are some which depend on GEOS libraries.

**Comment by sanAkel** [comment]
> To generate a coupled model grid, we use a whole bunch of programs. Some of these programs are stand alone, but there are some which depend on GEOS libraries.

@yvikhlya would please **list them**  and their **exact** dependencies. Thanks! Please also coordinate with @zhaobin74 to include sea ice and of course don't forget the (river) routing. 

@lcandre2 please add to this list your glacier routing routines as well. Thanks!

@tclune everything from the ocean model: MOM6 is entire outside of GEOS and is _standalone_, I believe same applies to CICE6, our _future_ sea ice module. AFAIK what @yvikhlya is referring to ⏫ is all for **coupling** with GEOS. The reason for including this aspect (of connections to GEOS) is with the thought of having clean **data drivers** for our _external_ components, e.g., MOM6, CICE6, etc.

**Comment by tclune** [comment]
@gmao-rreichle  I understand that these programs have dependencies on things that _must_ live in the main software repositories.    Hence the conflicting concerns.      If moved into the data repository, there would also be a bit of CMake logic that would do the necessary linking from in installation of GEOS.     Not a great situation maybe, but also not intractable.   Certainly the solution you have chosen is fine for the reasonably foreseeable future.     I would just want to revisit once the data repo plans firm up a bit.

**Comment by yvikhlya** [comment]
> would please list them and their exact dependencies

Oh, don't ask me please, because I have no idea what those dependencies are.

**Comment by zhaobin74** [comment]
> @yvikhlya would please **list them** and their **exact** dependencies. Thanks! Please also coordinate with > @zhaobin74 to include sea ice and of course don't forget the (river) routing.

cice grid generator doesn't have any dependencies except for the baselibs.

As @sanAkel  mentioned, moving forward we 'll have a CICE6 repo which contains all the utilities for grid and restart generation. The current set of tools in mk_restart will be carried for legacy purpose.

**Comment by sanAkel** [comment]
> > would please list them and their exact dependencies
> 
> Oh, don't ask me please, because I have no idea what those dependencies are.

@yvikhlya since you brought up 

```
Some of these programs are stand alone, but there are some which depend on GEOS libraries.
```
I assumed you are well aware of it. In any case, I feel it is not impossible to `grep use module` from the head of the `.F90`! For instance, I can list following:
https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/e0cf9fabbe700c3e23571466f444973dd6b20e4e/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts/SaltImpConverter.F90#L3-L7

**Comment by yvikhlya** [comment]
> I assumed you are well aware of it

No, I don't. I have no idea how the tile file, routing table and land BCs are generated.

**Comment by smahanam** [comment]
OK we are talking about mk_restarts:

 2 reasons why they all should stay together:
 1) all programs use the same master_til_file reader and getids function within the same directory
 2) In 2019, I made some changes to getids, WJ too made some changes later. We did not know what mk_ice, mk_saltice are doing, but we saw there are other programs that use the same subroutines, so we modified the calls in those programs too.
 They use MAPL, in fact, they should be using more from science modules, there are many instances that are duplicates to some science routines.
 
 
 Raster:
 It is a bit long story, I'll get back to you all later today.
 
 Tom, I am not going to be inconvenienced if you decide moving all these somewhere if a better long term option.  I withdraw all my comments above, if the changes are needed to handle these rapidly evolving improvements (we land added 4 programs and revamped an old one in last 4 years), I see mk_ice etc are new too.

**Comment by yvikhlya** [comment]
> OK we are talking about mk_restarts

Yes, exactly. @sanAkel we have a number of things with generating a coupled set up which don't go smoothly and need to be addressed, but let's not bring them up here. This is a subject for a separate issue (a big one).

**Comment by sanAkel** [comment]
> > OK we are talking about mk_restarts
> 
> Yes, exactly. @sanAkel we have a number of things with generating a coupled set up which don't go smoothly and need to be addressed, but let's not bring them up here. This is a subject for a separate issue (a big one).

okay I look forward to seeing that with its gorgeous details! 🌞

**Comment by smahanam** [comment]
Hi Tom,

For the proposed data repository - this is the thread you asked people for their input.

GCM reads few input files from $BCSDIR/$BCRSLV/. 
I am describing the background of the files those I am involved (from the land group).
I'll use /discover/nobackup/ltakacs/bcs/Icarus-NLv3/Icarus-NLv3_MERRA-2/CF0090x6C_DE1440xPE0720/ directory as an example.

Programs in GEOSsurface_GridComp/Shared/Raster created below files and 'rst', 'til','clsm' subdirectories.
CF0090x6C_DE1440xPE0720-Pfafstetter.til
green_clim_90x540.data
lai_clim_90x540.data
lnfm_clim_90x540.data 
ndvi_clim_90x540.data 
nirdf_90x540.dat 
vegdyn_90x540.dat
visdf_90x540.dat
irrigation_90x540.dat (future)

GCM reads above 9 files directly, while programs in mk_restarts, in addition to writing initial model state, read parameter 
files in 'clsm' subdirectory and write them also into catch(cn)_internal_rst files for the GCM to read.

I am not talking about topo_DYN_ave_90x540.data, topo_GWD_var_90x540.data and topo_TRB_var_90x540.data - I think it is Ben who created those.

We use various global data sets to derive those parameter files. As an example, here are the steps we follow to produce
lai_clim_90x540.data climatology file from MODIS MCD15A2H v6 LAI data.

STEP 1: Preprocessing - This step is usually performed in local machines. I guess, this step can continue to stay in local machines.
download MODIS for all years (2002-2020, 46 days per year, ~315 files per day), reproject to lat/lon (from 500m sinosoidal to 15 arcsec), 
stitch together, compute climatology on 15 arcsec pixels (46 days per year) etc. and prepare 'Analysis_Ready' global data files.

STEP 2: As of now /gpfsm/dnb42/projects/p16/ssd/land/l_data/LandBCs_files_for_mkCatchParam/V001 is our Analysis_Ready data
directory.
Programs in GEOSsurface_GridComp/Shared/Raster have data readers for files we create in Step 1. They read those files and
create fields and parameters on land tiles and write GCM input files  in /discover/nobackup/ltakacs/bcs/Icarus-NLv3/Icarus-NLv3_MERRA-2/CF0090x6C_DE1440xPE0720/

Diskspace:
Icarus-NLv3 $BCSDIR (including all grid resolutions) 1021 GB
Analysis_Ready directory: 468 GB

There is about 1.5 TB of another Analysis_Ready data set elsewhere that may be used for MERRA3. We have outsourced them to 617.0 and then to NCCS.  

Above is the workflow for land files. Let me know if you have questions.

Thanks,
Sarith
